plugin/helpers_bots.go

```diff
@@ -72,3 +105,44 @@ func (p *HelpersImpl) EnsureBot(bot *model.Bot) (retBotId string, retErr error)
 
 	return createdBot.UserId, nil
 }
+
+func (p *HelpersImpl) ShouldProcessMessage(post *model.Post, botUserId string, options ...ShouldProcessMessageOption) (bool, error) {
+	messageProcessOptions := &shouldProcessMessageOptions{}
+	for _, option := range options {
+		option(messageProcessOptions)
+	}
+
+	if !messageProcessOptions.AllowSystemMessages {
+		if post.IsSystemMessage() {
+			return false, nil
+		}
+	}
+
+	if !utils.StringInSlice(post.ChannelId, messageProcessOptions.FilterChannelIDs) {
```

Comment:
I think you and @hanzei synced up on this already, but I think we nee to distinguish an empty `FilterChannelIDs` from a populated one.